### PR TITLE
Turn pkg/apis into go module for easier import without our fork

### PR DESCRIPTION
**Pull request by sttts**
TODOs:
- [x] fix client generation
- [x] fix openapi generation

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -26,12 +26,14 @@ fi
...
 # Update generated CRD YAML
-${CONTROLLER_GEN} \
+cd pkg/apis
+../../${CONTROLLER_GEN} \
     crd \
     rbac:roleName=manager-role \
     webhook \
-    paths="./pkg/apis/..." \
-    output:crd:artifacts:config=config/crds
+    paths="./..." \
```
Comment (ncdc) [inline hack/update-codegen.sh:34]: Why was this commit/change needed?

File: go.mod [hunk 2]
```diff
@@ -22,23 +22,24 @@ require (
 	go.uber.org/multierr v1.7.0
 	google.golang.org/grpc v1.40.0
 	google.golang.org/protobuf v1.27.1
-	k8s.io/api v0.0.0
-	k8s.io/apiextensions-apiserver v0.0.0
-	k8s.io/apimachinery v0.23.5
+	k8s.io/api v0.24.1
+	k8s.io/apiextensions-apiserver v0.24.1
+	k8s.io/apimachinery v0.24.1
```
Comment (ncdc) [inline go.mod:27]: Even though this is a separate go module, we should match 0.23.5 that kcp core is using, right?

File: go.mod [hunk 3]
```diff
@@ -22,23 +22,24 @@ require (
...
 	k8s.io/cli-runtime v0.0.0
 	k8s.io/client-go v0.0.0
 	k8s.io/code-generator v0.0.0
 	k8s.io/component-base v0.0.0
-	k8s.io/klog/v2 v2.30.0
+	k8s.io/klog/v2 v2.60.1
 	k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65
 	k8s.io/kubernetes v1.23.5
-	k8s.io/utils v0.0.0-20211208161948-7d6a63dca704
+	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
```
Comment (ncdc) [inline go.mod:36]: Probably undo?

File: pkg/apis/go.mod [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+go 1.17
+
+require (
+	github.com/google/go-cmp v0.5.5
+	github.com/kcp-dev/logicalcluster v1.0.0
+	github.com/onsi/gomega v1.10.1
+	k8s.io/api v0.24.1
+	k8s.io/apiextensions-apiserver v0.24.1
+	k8s.io/apimachinery v0.24.1
```
Comment (ncdc) [inline pkg/apis/go.mod:11]: Should we match v0.23.5 from kcp core?

Comment (sttts) [reply pkg/apis/go.mod]: done

File: go.mod [hunk 5]
```diff
@@ -22,23 +22,24 @@ require (
...
+	k8s.io/api v0.24.1
+	k8s.io/apiextensions-apiserver v0.24.1
+	k8s.io/apimachinery v0.24.1
 	k8s.io/apiserver v0.0.0
 	k8s.io/cli-runtime v0.0.0
 	k8s.io/client-go v0.0.0
 	k8s.io/code-generator v0.0.0
 	k8s.io/component-base v0.0.0
-	k8s.io/klog/v2 v2.30.0
+	k8s.io/klog/v2 v2.60.1
```
[the remark this answers is not in the document]
Comment (sttts) [reply go.mod]: done

File: go.mod (hunk 2 above)
Comment (sttts) [reply go.mod]: This was a consequence of v0.24.1 in pkg/apis. Fixed.

**Pull request merged**